tests/test_docker_cat.py

```diff
@@ -0,0 +1,539 @@
+            assert sensor_line in output
+        # Wait for SonarQube to process the results
+        time.sleep(8)
+        # Check that the project was added to the server
+        output = requests.get(f"{cls.CAT_URL}/api/projects/search?projects={project_key}",
+                        auth=("admin", "admin")).json()
+        # Hint: if this test fails, the project is not on the server
+        assert output['components'][0]['key'] == project_key
+        # Get the number of issues of the project
+        output = requests.get(f"{cls.CAT_URL}/api/issues/search?componentKeys={project_key}",
+                        auth=("admin", "admin")).json()['issues']
+        issues = [ issue for issue in output if issue['status'] in ('OPEN', 'TO_REVIEW') ]
+        # Hint: if this test fails, there should be {nb_issues issues} on the {language_name} dummy project with the Sonar way QP but {len(issues)} were found
+        assert len(issues) == nb_issues
+        print("Analysis with Sonar way QP ran as expected.")
+        # If the language has a specific CNES Quality Profile, it must also be tested
+        if cnes_qp:
+            # Switch to CNES QP
+            requests.post(f"{cls.CAT_URL}/api/qualityprofiles/add_project",
+                auth=("admin", "admin"),
+                data={
+                    "language": language_key,
+                    "project": project_key,
+                    "qualityProfile": cnes_qp
+                })
+            # Rerun the analysis
+            docker_client.containers.get(cls.CAT_CONTAINER_NAME).exec_run(
+                f"sonar-scanner -Dsonar.host.url=http://localhost:9000 -Dsonar.projectBaseDir=/media/sf_Shared/tests/{folder}"
+                )
+            # Wait for SonarQube to process the results
+            time.sleep(8)
+            # Switch back to the Sonar way QP (in case the test needs to be rerun)
+            requests.post(f"{cls.CAT_URL}/api/qualityprofiles/add_project",
+                auth=("admin", "admin"),
+                data={
+                    "language": language_key,
+                    "project": project_key,
+                    "qualityProfile": "Sonar way"
+                })
+            # Get the new number of issues
+            output = requests.get(f"{cls.CAT_URL}/api/issues/search?componentKeys={project_key}",
+                auth=("admin", "admin")).json()['issues']
+            issues = [ issue for issue in output if issue['status'] in ('OPEN', 'TO_REVIEW') ]
+            # Hint: if this test fails, there should be {nb_issues_cnes_qp} issues on the {language_name} dummy project with the {cnes_qp} QP but {len(issues)} were found
+            assert len(issues) == nb_issues_cnes_qp
+
+    @classmethod
+    def analysis_tool(cls, tool: str, cmd: str, ref_file: str, tmp_file: str, store_output: bool = True):
+        """This function tests that the image can run a specified code analyzer
```

Comment:
Codacy found an issue: [1 blank line required between summary line and description (found 0) (D205)](https://app.codacy.com/gh/cnescatlab/docker-cat/pullRequest?prid=6253839)